File: examples/typescript/src/Person.ts
/**
 * A class representing a person.
 * @class Person
 * @constructor {string} name - The name of the person.
 * @constructor {number} age - The age of the person.
 * @constructor {string} employmentStatus - The employment status of the person.
 * 
 * @method getName - Returns the name of the person.
 * @method getAge - Returns the age of the person.
 * @method getEmployed - Returns the employment status of the person.
 * @method setName - Sets the name of the person.
 * @method setAge - Sets the age of the person.
 * @method setEmployed - Sets the employment status of the person.
 * @method toString - Returns a string representation of the person.
 */
export class Person {
  private name: string;
  private age: number;
  private employmentStatus: string;

  constructor(name: string, age: number, employmentStatus: string) {
    this.name = name;
    this.age = age;
    this.employmentStatus = employmentStatus;
  }

  getName(): string {
    return this.name;
  }

  getAge(): number {
    return this.age;
  }

  getEmployed(): string {
    return this.employmentStatus;
  }

  setName(name: string): void {
    this.name = name;
  }

  setAge(age: number): void {
    this.age = age;
  }

  setEmployed(employed: string): void {
    this.employmentStatus = employed;
  }

  toString(): string {
    return `Name: ${this.name}, Age: ${this.age}, Employed: ${this.employmentStatus}`;
  }
}

File: examples/python/Person.py
class Person:
  """
  A class representing a person.

  Attributes:
    name (str): The name of the person.
    age (int): The age of the person.
    employment_status (str): The employment status of the person.

  Methods:
    get_name(): Returns the name of the person.
    get_age(): Returns the age of the person.
    get_employed(): Returns the employment status of the person.
    set_name(name): Sets the name of the person.
    set_age(age): Sets the age of the person.
    set_employed(employed): Sets the employment status of the person.

  Overrides:
    __str__(): Returns a string representation of the person.
  """

  def __init__(self, name, age, employment_status):
    self.name = name
    self.age = age
    self.employment_status = employment_status

  def get_name(self):
    return self.name

  def get_age(self):
    return self.age

  def get_employed(self):
    return self.employed

  def set_name(self, name):
    self.name = name

  def set_age(self, age):
    self.age = age

  def set_employed(self, employed):
    self.employment_status = employed

  def __str__(self) -> str:
    return f"Name: {self.name}, Age: {self.age}, Employed: {self.employment_status}"  
